java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java

```diff
@@ -221,6 +227,37 @@ public HttpEventResponse handle(HttpEventRequest request) {
         }
     }
 
+    private void addProxyHeaders(HttpEventResponse.HttpEventResponseBuilder builder, ConfigService config) {
+        Stream<ConfigService.ConfigProperty> toCapture =
+            Stream.of(
+                ProxyConfigProperty.SOURCE_AUTH_STRATEGY_IDENTIFIER,
+                OAuthRefreshTokenSourceAuthStrategy.ConfigProperty.GRANT_TYPE
+                );
+
+        builder.header(ResponseHeader.SOURCE_AUTH.getHttpHeader(),
+            toCapture
+                .map(param -> Pair.of(param, config.getConfigPropertyAsOptional(param).orElse("none")))
+                .map(p -> p.getKey() + "=" + p.getValue())
+                .collect(Collectors.joining(";")));
+
+
+        Stream<ConfigService.ConfigProperty> toCaptureLastModified =
+            Stream.of(OAuthRefreshTokenSourceAuthStrategy.ConfigProperty.ACCESS_TOKEN);
+
+        builder.header(ResponseHeader.CONFIG_FILLED.getHttpHeader(),
+            toCaptureLastModified
+                .map(param -> Pair.of(param, config.getConfigPropertyWithMetadata(param)
+                        .map(ConfigService.ConfigValueWithMetadata::getLastModifiedDate)
+                        .map(lastModified -> lastModified.map(Instant::toString)
+                                            .orElse("not-available") // config service that doesn't support it
+                        )
+                        .orElse("")))
+                .map(p -> p.getKey() + "=" + p.getValue())
+                .collect(Collectors.joining(";")));
+
+        //q: make this all conditional on some control header being passed (e.g. `X-Psoxy-Debug-Config`)?
```

Comment:
I was about to comment this, no real need to do a full request, just configuration.
We should include more stuff in the response
- proxy version (should be a constant in the deployment somehow)
- basic config check on all parameters, list all and return true/false if blank, so we can quickly see misconfigurations.

Reply:
expanded a bit; but w/o reflection, there's not a full list of ConfigParameters anywhere.

we could dump keys from env, but would likely be large and potentially unsafe to do so (if someone adds something weird to env on all their lambdas or something).